File: semgrep/qa-testing-rules.java
// Semgrep fixture tests for qa-testing-rules Java rules.

import org.junit.jupiter.api.Disabled;

class ExampleTest {
  void sleepBad() throws Exception {
    // ruleid: qa-no-sleep-in-tests-java
    Thread.sleep(100);
  }

  // ruleid: qa-no-disabled-tests-java
  @Disabled
  void disabledBad() {}
}

File: semgrep/qa-testing-rules.rs
// Semgrep fixture tests for qa-testing-rules Rust rules.

#[test]
fn sleep_bad() {
    // ruleid: qa-no-sleep-in-tests-rust
    std::thread::sleep(std::time::Duration::from_millis(100));
}

#[test]
// ruleid: qa-no-ignored-tests-rust
#[ignore]
fn ignored_bad() {}
